File: JavaScript/Factorialusingclass.js
class Factorial {
    constructor(n) {
      this.n = n;
    }
  
    calculate() {
      if (this.n < 0) {
        throw new Error('Factorial is not defined for negative numbers');
      }
      return this._factorial(this.n);
    }
  
    _factorial(num) {
      if (num <= 1) return 1;
      return num * this._factorial(num - 1);
    }
  }
  
  // Sample input
  const number = 5;
  const factorialObj = new Factorial(number);
  console.log(`Factorial of ${number} is: ${factorialObj.calculate()}`);
  

File: Python/factorial_using_class.py
class FactorialCalculator:
    def __init__(self, number):
        self.number = number

    def calculate_factorial(self):
        if self.number < 0:
            return "Factorial is not defined for negative numbers."
        elif self.number == 0 or self.number == 1:
            return 1
        else:
            result = 1
            for i in range(2, self.number + 1):
                result *= i
            return result

# Example usage:
if __name__ == "__main__":
    num = 6
    factorial_calculator = FactorialCalculator(num)
    result = factorial_calculator.calculate_factorial()
    print(f"The factorial of {num} is: {result}")
